index.ts

```diff
@@ -407,17 +407,22 @@ class DatabaseFake {
     tableName: string,
     callback: (doc: GenericDocument) => void,
   ) {
-    for (const write of Object.values(this._writes)) {
-      if (write.isInsert) {
-        const document = write.newValue;
-        if (tableNameFromId(document._id) === tableName) {
-          callback(document);
+    for (let i = 0; i < this._writes.length; i++) {
```

Comment:
Seems like there's a bug around if there's an insert at one level and a delete / update at a lower level. Regression test + fix?